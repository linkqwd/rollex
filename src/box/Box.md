Simple component which can accept props to add margin and padding


```js
import { Box } from 'zk-ui';

<>
    <Box mb='30px'>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus ea excepturi expedita illum nisi porro
        quas qui quia quo tempore! Enim esse maiores repellat soluta. Aliquam exercitationem ipsum omnis quia.
    </Box>
    <Box mb='30px' p='15px'>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus ea excepturi expedita illum nisi porro
        quas qui quia quo tempore! Enim esse maiores repellat soluta. Aliquam exercitationem ipsum omnis quia.
    </Box>
    <Box mb='30px'>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus ea excepturi expedita illum nisi porro
        quas qui quia quo tempore! Enim esse maiores repellat soluta. Aliquam exercitationem ipsum omnis quia.
    </Box>
</>
```

```js
import { Row, Col, Box } from 'zk-ui';

<Row>
    <Col grow={1}>
        <Box p='15px'>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus ea excepturi expedita illum nisi porro
            quas qui quia quo tempore! Enim esse maiores repellat soluta. Aliquam exercitationem ipsum omnis quia.
        </Box>
    </Col>
    <Col grow={1}>
        <Box p='15px'>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus ea excepturi expedita illum nisi porro
            quas qui quia quo tempore! Enim esse maiores repellat soluta. Aliquam exercitationem ipsum omnis quia.
        </Box>
    </Col>
    <Col grow={1}>
        <Box p='15px'>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ducimus ea excepturi expedita illum nisi porro
            quas qui quia quo tempore! Enim esse maiores repellat soluta. Aliquam exercitationem ipsum omnis quia.
        </Box>
    </Col>
</Row>
```
