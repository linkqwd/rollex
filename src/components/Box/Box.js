import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';


const BoxStyles = styled.div`
    ${({ m }) => m && css`margin: ${m};`}
    ${({ mt }) => mt && css`margin-top: ${mt};`}
    ${({ mr }) => mr && css`margin-right: ${mr};`}
    ${({ mb }) => mb && css`margin-bottom: ${mb};`}
    ${({ ml }) => ml && css`margin-left: ${ml};`}
    ${({ mx }) => mx && css`margin-left: ${mx}; margin-right: ${mx};`}
    ${({ my }) => my && css`margin-top: ${my}; margin-bottom: ${my};`}

    ${({ p }) => p && css`padding: ${p};`}
    ${({ pt }) => pt && css`padding-top: ${pt};`}
    ${({ pr }) => pr && css`padding-right: ${pr};`}
    ${({ pb }) => pb && css`padding-bottom: ${pb};`}
    ${({ pl }) => pl && css`padding-left: ${pl};`}
    ${({ px }) => px && css`padding-left: ${px}; padding-right: ${px};`}
    ${({ py }) => py && css`padding-top: ${py}; padding-bottom: ${py};`}
`;

export const Box = ({ children, dataQa, ...rest }) => {
    return (
        <BoxStyles data-qa={dataQa} {...rest}>
            {children}
        </BoxStyles>
    );
};

Box.propTypes = {
    children: PropTypes.node,
    /** selector for QA needs */
    dataQa: PropTypes.string,
    /** margin: 5px 10px 15px 20px; or margin: 5px; */
    m: PropTypes.string,
    /** margin-top: 5px; */
    mt: PropTypes.string,
    /** margin-right: 10px; */
    mr: PropTypes.string,
    /** margin-bottom: 15px; */
    mb: PropTypes.string,
    /** margin-left: 20px; */
    ml: PropTypes.string,
    /** margin-left: 10px; margin-right: 10px; */
    mx: PropTypes.string,
    /** margin-top: 10px; margin-bottom: 10px; */
    my: PropTypes.string,
    /** padding: 5px 10px 15px 20px; */
    p: PropTypes.string,
    /** padding-top: 5px; */
    pt: PropTypes.string,
    /** padding-right: 10px; */
    pr: PropTypes.string,
    /** padding-bottom: 15px; */
    pb: PropTypes.string,
    /** padding-left: 20px; */
    pl: PropTypes.string,
    /** padding-left: 10px; padding-right: 10px; */
    px: PropTypes.string,
    /** padding-top: 10px; padding-bottom: 10px; */
    py: PropTypes.string,
};
