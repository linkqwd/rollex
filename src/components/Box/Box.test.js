import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import 'jest-styled-components';

import { Box } from './Box';


describe('Test Box component', () => {
    it('should render Box with children and props', () => {
        const wrapper = mount(
            <Box p='15px'><span className='test'>test</span></Box>
        );
        expect(wrapper.exists('div')).toEqual(true);
        expect(wrapper.contains(<span className='test'>test</span>)).toEqual(true);
        expect(wrapper.props().p).toEqual('15px');
    });

    it('renders correctly', () => {
        const rendered = create(<Box />).toJSON();
        expect(rendered).toMatchSnapshot();
    });

    it('it applies styles according to passed props', () => {
        const rendered = create(<Box mb='30px' />).toJSON();
        expect(rendered).toHaveStyleRule('margin-bottom', '30px');
    });
});
