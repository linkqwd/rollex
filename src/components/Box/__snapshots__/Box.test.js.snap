// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Box component renders correctly 1`] = `
<div
  className=""
/>
`;
